packages/sites/clinepi-site/webapp/js/client/component-wrappers/SiteHeader.jsx

```diff
@@ -26,10 +26,37 @@ export default function SiteHeaderWrapper() {
 
     const { diyDatasets, reloadDiyDatasets } = useDiyDatasets();
 
+    // for now, we default to each studies section being open and the useEffect
+    // ensures that the sections open when searching
+    const [expandUserStudies, setExpandUserStudies] = useState(true);
+    const [expandCuratedStudies, setExpandCuratedStudies] = useState(true);
+    useEffect(() => {
+      if (searchTerm && searchTerm.length > 0) {
+        setExpandUserStudies(true);
+        setExpandCuratedStudies(true);
+      }
+    }, [searchTerm, setExpandUserStudies, setExpandCuratedStudies]);
```

Comment:
Instead of using a `useEffect` hook here, you can make a callback function that calls `setSearchTerm` and both `setExpand*Studies` functions. I think it's a little bit clearer, and it will also remove a render cascade: calling a `setState` setter in a `useEffect` callback, due to a different `useState` value changing, will incur two renders, since the `useEffect` callback will be called _after_ the first state has been updated.